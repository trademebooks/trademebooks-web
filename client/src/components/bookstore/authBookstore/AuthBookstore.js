import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { MDBRow, MDBCol, MDBContainer, MDBBtn, MDBIcon } from 'mdbreact'
import { toastr } from 'react-redux-toastr'

import Books from '../../books/common/Books'
import config from '../../../config'

import './AuthBookstore.scss'

const AuthBookstore = ({
  auth: {
    user: { username }
  },
  books,
  getBookstoreByUsername
}) => {
  const [clipBoardUrl, setClipBoardUrl] = useState('')

  useEffect(() => {
    getBookstoreByUsername('auth')
  }, [getBookstoreByUsername])

  useEffect(() => {
    setClipBoardUrl(`${config.BASE_URL}/bookstore/${username}`)
  }, [username])

  const handleUrlCopyToClipboard = (e) => {
    navigator.clipboard.writeText(clipBoardUrl).then(
      function () {
        toastr.success('Your bookstore Link has been copied to your clipboard!')
      },
      function () {
        toastr.error('Clipboard write failed.')
      }
    )
  }

  return (
    <>
      <div className="header-container text-center">
        <h3 className="font-weight-bold">My Bookstore</h3>
        <div>
          <div className="bookstore-clipboard-container">
            <div className="bookstore-clipboard-inner-container">
              <div className="bookstore-clipboard__top-section">
                Copy &amp; Share your Bookstore Link!
              </div>
              <div className="bookstore-clipboard__bottom-section">
                <input
                  className="form-control bookstore-clipboard__bottom-section-input-text"
                  readOnly
                  type="text"
                  value={clipBoardUrl}
                />
                <MDBBtn
                  size="sm"
                  className="bookstore-clipboard__bottom-section-copy-button"
                  onClick={handleUrlCopyToClipboard}
                >
                  <MDBIcon icon="clipboard" size="lg" />
                </MDBBtn>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div className="mt-4">
        <MDBContainer>
          <MDBRow center={true}>
            <MDBCol sm="12">
              {books.length < 1 ? (
                <div>There are no books in this bookstore...</div>
              ) : (
                <Books books={books} editFlag={true} />
              )}
            </MDBCol>
          </MDBRow>
        </MDBContainer>
      </div>
    </>
  )
}

AuthBookstore.propTypes = {
  books: PropTypes.array.isRequired,
  getBookstoreByUsername: PropTypes.func.isRequired
}

export default AuthBookstore
