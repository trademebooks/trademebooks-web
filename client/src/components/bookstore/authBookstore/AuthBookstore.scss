.bookstore-clipboard {
  &-container {
    margin-top: 1rem;
  }

  &-inner-container {
    width: 60%;
    margin: 0 auto;

    @media screen and (min-width: 1200px) /*and (max-width: 1200px) */ {
      width: 50%;
    }

    @media screen and (min-width: 767px) and (max-width: 1200px) {
      width: 70%;
    }

    @media screen and (max-width: 767px) {
      width: 95%;
    }
  }

  &__top-section {
    color: #898989;
  }

  &__bottom-section {
    display: flex;

    &-input-text {
      border-radius: 0.3rem 0 0 0.3rem !important;

      &[readonly] {
        background-color: white !important;
      }
    }

    &-copy-button {
      margin: 0 !important;
      border-radius: 0 0.3rem 0.3rem 0 !important;
    }
  }
}
