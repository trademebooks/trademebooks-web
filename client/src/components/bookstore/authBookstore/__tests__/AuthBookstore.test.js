import React from 'react'
import renderer from 'react-test-renderer'
import { render } from '@testing-library/react'

import AuthBookstore from '../AuthBookstore'

describe('Component: AuthBookstore', () => {
  const props = {
    auth: {
      user: {
        username: 'johndoe'
      }
    },
    books: [],
    getBookstoreByUsername: jest.fn()
  }

  it('should match the snapshot', () => {
    const authBookstoreSnapshot = renderer
      .create(<AuthBookstore {...props} />)
      .toJSON()
    expect(authBookstoreSnapshot).toMatchSnapshot()
  })

  it('should render properly', () => {
    const authBookstore = render(<AuthBookstore {...props} />)
  })
})
