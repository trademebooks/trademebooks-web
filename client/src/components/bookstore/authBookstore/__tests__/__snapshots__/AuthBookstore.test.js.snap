// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: AuthBookstore should match the snapshot 1`] = `
Array [
  <div
    className="header-container text-center"
  >
    <h3
      className="font-weight-bold"
    >
      My Bookstore
    </h3>
    <div>
      <div
        className="bookstore-clipboard-container"
      >
        <div
          className="bookstore-clipboard-inner-container"
        >
          <div
            className="bookstore-clipboard__top-section"
          >
            Copy & Share your Bookstore Link!
          </div>
          <div
            className="bookstore-clipboard__bottom-section"
          >
            <input
              className="form-control bookstore-clipboard__bottom-section-input-text"
              readOnly={true}
              type="text"
              value=""
            />
            <button
              className="btn-default btn Ripple-parent btn-sm bookstore-clipboard__bottom-section-copy-button"
              data-test="button"
              onClick={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <i
                className="fa fa-clipboard fa-lg"
                data-test="fa"
              />
              <div
                className="Ripple "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  <div
    className="mt-4"
  >
    <div
      className="container"
      data-test="container"
    >
      <div
        className="row justify-content-center"
        data-test="row"
      >
        <div
          className="col-sm-12"
          data-test="col"
        >
          <div>
            There are no books in this bookstore...
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
